server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -297,6 +298,28 @@ public Consumer lockAndLoad(Consumer c) {
 
     }
 
+    /**
+     * Find a consumer by uuid and immediately lock it.
+     *
+     * @param consumerUuid the uuid of the target consumer.
+     *
+     * @return the Consumer matching the given uuid, null if the consumer was not found.
+     */
+    @Transactional
```

Comment:
> does not make sense to @Transactional-ize the method that's sole purpose is to lock an entity.

Why not, as this would be the first time the entity is looked up. It is different than the lockAndLoad method as the consumer was expected to already have been looked up in Larger transaction. Perhaps I'm missing something.